cyclonedx-bom/src/models/vulnerability_rating.rs

```diff
@@ -116,6 +114,34 @@ impl Validate for VulnerabilityRatings {
     }
 }
 
+/// Defines a vulnerability's score. According to the spec this is a decimal value.
+/// However, deriving `Eq` on `Bom` and all its subsequent data structures
+/// prevents us from using a float value for `Score`. Implementing `Score` as
+/// an i32 allows us to easily derive `Eq`. We have chosen a factor of 1000 to
+/// convert a f32 into i32 because OWASP's scoring method uses up to three decimal places.
+///
+/// Defined via the [XML schema](https://cyclonedx.org/docs/1.4/xml/#type_ratingType)
+#[derive(Debug, PartialEq, Eq)]
+pub struct Score(i32);
```

Comment:
There is a small crate called [`ordered-float`](https://github.com/reem/rust-ordered-float) that should cover this nicely. It provides implementations for `Ord` & `Eq`. The `Score` struct then can look like:

```rust
#[derive(Debug, PartialEq, Eq)]
pub struct Score(OrderedFloat<f32>);

impl Score {
    pub fn new_unchecked(score: f32) -> Self {
        Score(score.into())
    }

    pub fn from_f32(score: f32) -> Option<Self> {
        // or add range check here, maybe clamp between 0..=10?
        Some(Score(score.into()))
    }

    pub fn to_f32(&self) -> f32 {
        self.0.0
    }
}

impl From<f32> for Score {
    fn from(value: f32) -> Self {
        Score(value.into())
    }
}

impl From<Score> for f32 {
    fn from(value: Score) -> f32 {
        value.0.0
    }
}
```

This simplifies conversions & should provide easier handling of scores.